tools/serverpod_cli/lib/src/analyzer/entities/entity_dependency_resolver.dart

```diff
@@ -262,7 +263,7 @@ class EntityDependencyResolver {
 
     if (relation.name == null) {
       var foreignFieldName =
-          '_${classDefinition.tableName}_${fieldDefinition.name}_${classDefinition.tableName}Id';
+          '_${classDefinition.tableName}${fieldDefinition.name.toCamelCase()}${classDefinition.tableName?.toCamelCase()}Id';
```

Comment:
Suggestion: `tableName` here is allowed to be snake_case, convert the table name to camel case as well!

Remove the first table name. You decide!

Reply:
Create a test case for this too!